Common/src/geometry/CGeometry.cpp

```diff
@@ -4149,3 +4110,58 @@ void CGeometry::SetNaturalElementColoring()
   /*--- In parallel, set the group size to nElem to protect client code. ---*/
   if (omp_get_max_threads() > 1) elemColorGroupSize = nElem;
 }
+
+void CGeometry::ComputeWallDistance(CConfig **config_container, CGeometry ****geometry_container){
+
+  int nZone = config_container[ZONE_0]->GetnZone();
+  bool allEmpty = true;
+  vector<bool> wallDistanceNeeded(nZone, false);
+
+  for (int iInst = 0; iInst < config_container[ZONE_0]->GetnTimeInstances(); iInst++){
+    for (int iZone = 0; iZone < nZone; iZone++){
+
+      /*--- Check if a zone needs the wall distance and store a boolean ---*/
+
+      ENUM_MAIN_SOLVER kindSolver = static_cast<ENUM_MAIN_SOLVER>(config_container[iZone]->GetKind_Solver());
+      if (kindSolver == RANS ||
+          kindSolver == INC_RANS ||
+          kindSolver == DISC_ADJ_RANS ||
+          kindSolver == DISC_ADJ_INC_RANS ||
+          kindSolver == FEM_LES ||
+          kindSolver == FEM_RANS){
+        wallDistanceNeeded[iZone] = true;
+      }
+
+      /*--- Set the wall distances in all zones to the numerical limit.
+     * This is necessary, because before a computed distance is set, it will be checked
+     * whether the new distance is smaller than the currently stored one. ---*/
+      CGeometry *geometry = geometry_container[iZone][iInst][MESH_0];
+      if (wallDistanceNeeded[iZone])
+        geometry->SetWallDistance(numeric_limits<su2double>::max());
+    }
+
+    /*--- Loop over all zones and compute the ADT based on the viscous walls in that zone ---*/
+    for (int iZone = 0; iZone < nZone; iZone++){
+      CGeometry *geometry = geometry_container[iZone][iInst][MESH_0];
+      unique_ptr<CADTElemClass> WallADT = geometry->ComputeViscousWallADT(config_container[iZone]);
+      if (WallADT && !WallADT->IsEmpty()){
```

Comment:
Is it ok to do this? Don't you need two nested if's to avoid potentially dereferencing a nullptr?

Reply:
Yep that is totally legal, even written in the standard. If the first operand of && is false then the second one is not evaluated. See http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2010/n3092.pdf page 118/119